### PR TITLE
Fix issue with block bindings in while/do-while loops

### LINKED ISSUE #1970: TypeError: Cannot read property 'transform' of undefined
I receive `TypeError: Cannot read property 'transform' of undefined` when trying to compile any code that follows this pattern:

```
while( true ) {
    let foo = 1;
    ( () => { foo; } )();
}
```

It seems to involve the following components:

1. `while` loop (`for` loops seem unaffected)
2. variable declared within loop using either `let` or `const` (`var` is fine)
3. variable created in `2` is referenced inside a closure within the loop

I understand this code might be buggy or bad practice, but the error from Traceur is cryptic.

I am compiling using traceur v0.0.90 via grunt-traceur v0.5.4.

### PR DESCRIPTION
**Pull request by arv**
The transformer was incorrectly calling transformAny with undefined.

Fixes #1970

### CODE REVIEW

**Comment by arv** [comment]
@johnjbarton PTAL

File: src/codegeneration/BlockBindingTransformer.js [hunk 1]
```diff
@@ -590,29 +590,30 @@ export class BlockBindingTransformer extends ParseTreeTransformer {
...
+                return new VariableDeclaration(null,
+                    bindingIdentifier, null, declaration.initializer);
+              }));
 
-        initializer = renameAll(renames, initializer);
-      } else {
-        initializer = this.transformAny(tree.initializer);
+          initializer = renameAll(renames, initializer);
+        } else  {
+          initializer = this.transformAny(tree.initializer);
```
Comment (johnjbarton) [inline src/codegeneration/BlockBindingTransformer.js]: Before we would end up here for `var`. But the fail case is `let`. So the tree must have been transformed from `let` -> `var` before we get here.  Is that correct?

**Comment by johnjbarton** [comment]
LGTM, but I don't see immediately how the fix and the bug are connected.

**Comment by arv** [comment]
A while back I made the transformers and visitors more strict. They now fail when `undefined` is passed which was the case because `WhileStatement` does not have an `initializer`.

**Pull request merged**